packages/EasyCodingStandard/packages/ChangedFilesDetector/src/ChangedFilesDetector.php

```diff
@@ -23,10 +22,17 @@
      */
     private $fileHashComputer;
 
-    public function __construct(CacheFactory $cacheFactory, FileHashComputer $fileHashComputer)
+    public function __construct(FileHashComputer $fileHashComputer)
     {
-        $this->cache = $cacheFactory->create();
         $this->fileHashComputer = $fileHashComputer;
+    }
+
+    public function setCache(?Cache $cache): void
```

Comment:
Optional setter are very bad practice, since they can change behavior on the fly.
But also do nothing even when used.

Please keep constructor